### PR TITLE
[YUNIKORN-2727] Fix Dead Links and Update readme for Docusaurus v3

### PR DESCRIPTION
**Pull request by ryankert01**
Fix Dead Links and Update readme for Docusaurus v3

#### Issue 1: Dead Link in "Deploy the Scheduler"
- Problem: Dead link at example in "Deploy the Scheduler" section.
- Current: https://yunikorn.apache.org/docs/developer_guide/deployment/#deploy-the-admission-controller
- Solution: Replace with correct links:
https://yunikorn.apache.org/docs/next/developer_guide/deployment/#deploy-the-scheduler
https://yunikorn.apache.org/docs/next/developer_guide/deployment/#Deploy-the-Scheduler
- Cause: Migration to Docusaurus v3 with strict URL regulations.

#### Issue 2: Outdated Docusaurus Version in README
- Problem: README mentions Docusaurus v2.
- Current: "The website is built based using docusaurus-v2."
- Solution: Update to v3.
- New: "The website is built based using Docusaurus v3."

### What type of PR is it?
* [ ] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [x] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2727

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -9251,4 +9251,4 @@ packages:
 
   /zwitch@2.0.4:
     resolution: {integrity: sha512-bXE4cR/kVZhKZX/RjPEflHaKVhUVl85noU3v6b8apfQEc1x4A+zBxjZ4lN8LqGd6WZ3dl98pY4o717VFmoPp+A==}
-    dev: false
+    dev: false
```
Comment (chia7712) [inline pnpm-lock.yaml:9254]: I guess this change is unnecessary, right?

**Review by chia7712 — COMMENTED** [review]
@ryankert01 thanks for this patch. one small comment is left

File: pnpm-lock.yaml (hunk 1 above)
Comment (ryankert01) [reply pnpm-lock.yaml:9254]: It looks identical to me. It's caused by me pulling back changes on [pnpm-lock.yaml](https://github.com/apache/yunikorn-site/pull/451/commits/8989cd5a598965b08b46c95bb2eceec065e378de#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bb) on my first commit(caused by normal pnpm compilation). Also, wanna ask should I preserve those changes caused by normal compilation?

Comment (chia7712) [reply pnpm-lock.yaml:9254]: normally, we should avoid unnecessary changes. Hence, please revert it

Comment (ryankert01) [reply pnpm-lock.yaml]: Done.

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request closed without merging**